### ISSUE TITLE
Custom array for `rule-properties-order` should respect long-hand order.

### ISSUE THREAD

**Issue by MadLittleMods** [stylelint/stylelint#373]
### [`rule-properties-order`](https://github.com/stylelint/stylelint/blob/master/src/rules/rule-properties-order/README.md)

Maybe all of this should just be delegated to a separate module that can expand the string array to include the long-hand versions.

---

Say you specify an order of:
```js
postcss([
	stylelint({
		rules: {
			'rule-properties-order': [2, [
					'border',
					'color',
				]
			]
		}
	})
])
```

It should complain about:

```css
.foo {
    color: #f00;
    border-bottom: 2px solid #000;
}
```

**note:** For this example, we would have to be careful of `border-radius` as that is a separate property.

---

We can also extrapolate from the `top`, `left`, `bottom`, `right` declarations for the order of other properties with those sort of long-hands.

```
[
	'top',
	'left',
	'bottom',
	'right'

	'border'
]
```

Would enforce the order of the long-hand border properties as well.

```
.foo {
	border-top: 1px solid #f00;
	border-left: 1px solid #f00;
	border-bottom: 1px solid #f00;
	border-right: 1px solid #f00;
}
```

**Comment by davidtheclark** [comment]
Thanks, @MadLittleMods: I think that's a good enhancement idea, and should be easy to implement.

**Comment by MadLittleMods** [comment]
Also to reiterate what I mentioned in [#374](https://github.com/stylelint/stylelint/issues/374#issuecomment-139503617):

> I suggest we make some sensible defaults instead. For example people always specify border this way in shorthand even though it doesn't matter.
> ```
border-width: 3px;
border-style: solid;
border-color: #f00;
```

**Comment by davidtheclark** [comment]
In #391 I decided to leave the order of extension properties arbitrary unless the user has specified it.

stylelint has deliberately been very, very light on defaults, so I didn't want to insert one as opinionated as the ordering of, say, `border-*` properties. Users who want `border-*` properties ordered in a specific way can do so by specifying the order they want themselves.

**Comment by DanGamble89** [comment]
In respect to this would it also be able to implement nested arrays like CSSComb's sort order as my code style tends to be like:

```
  display: block;
  padding-top: 6px;
  padding-right: 10px;
  padding-bottom: 6px;
  padding-left: 10px;
  width: 100%;

  line-height: inherit;

  appearance: none;
  border: 1px solid transparent;
  color: map-get($colors, text);
```

**Comment by davidtheclark** [comment]
@DanGamble89 : This sounds like a separate feature request, for nested arrays that would enforce empty newlines above specified properties. Is that right, or am I misunderstanding? If so, please open a new issue for this distinct feature.

**Comment by DanGamble89** [comment]
Yes that is correct, sure thing i plopped it here just as i saw there was an open issue regarding sort order.

**Comment by MadLittleMods** [comment]
@davidtheclark I haven't tested but it seems this doesn't take into account things like `border` vs `border-radius` which are separate and not long-hand properties. Any definitive decision either way?

**Comment by davidtheclark** [comment]
@MadLittleMods : You're right it does not. I think the current does what it can without having to maintain lists of specific CSS properties that behave differently. I was thinking that trying to maintain property lists like that might be a dark and dangerous road to head down ... but open to a PR if you or someone else can pull it off effectively.
